### PR TITLE
feat: Implement Game Score Sidebar and Score Storage(GlobalStorage in VSC) Infrastructure with Score Transmission Logic

### PR DESCRIPTION
**Pull request by Ooyd**
# Pull Request Template

## Type of PR

- [x] Feature - Platform
- [ ] Feature - Game
- [ ] Fix - Bug

## Description of PR
This PR introduces a new feature for the game platform: a score sidebar and an associated scoring system.
It includes the creation of a directory and files for score storage, and implements logic for transmitting scores form the index.

## Changes Made
- Developed a game score sidebar to display player scores during gameplay.
- Create a new folder(VSC GlobalStroage) and file structure for storing game scores.

## Testing Method
- Manual testing was conducted to ensure the score sidebar updates in real-time.
- Integration tests checked the interaction between the score system and other game components.

## Related Issues
- #17 
- #23 

## Checklist

- [x] Prepared for code review (Assignees set, Labels applied, etc.)
- [x] Checked that changes do not negatively affect existing features
- [ ] Documentation completed for new features
- [ ] All unit tests passed
- [ ] Testing in the test environment completed
- [x] Linked this PR in the related issue
- [ ] Communicated sufficiently with the relevant team or person before submitting PR

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -1,52 +1,134 @@
...
+  }
+}
+
+// Game score generation function for local scoreboard(Use globalStorage)
+function saveScore(context: vscode.ExtensionContext, gameId: string, scoreData: { player: string, score: number }) {
+  // Create full path to file to save
+  const scoresPath = path.join(context.globalStorageUri.fsPath, `${gameId}-scores.json`);
+  console.log('scoresPath',scoresPath);
+  const directoryPath = path.dirname(scoresPath);
+  console.log('directoryPath',directoryPath);
```
Comment (junman95) [inline src/extension.ts]: erase console logs please!

File: src/extension.ts [hunk 2]
```diff
@@ -1,52 +1,134 @@
...
 
+// Function for creating directories
+function ensureDirectoryExistence(directoryPath: string){
+  if (!fs.existsSync(directoryPath)) {
+    fs.mkdirSync(directoryPath, { recursive: true});
+  }
+}
+
+// Game score generation function for local scoreboard(Use globalStorage)
+function saveScore(context: vscode.ExtensionContext, gameId: string, scoreData: { player: string, score: number }) {
```
Comment (junman95) [inline src/extension.ts]: Move this function to util!
Make sure it should be a pure functions(for testing and lesser side-effects)

File: src/extension.ts [hunk 3]
```diff
@@ -1,52 +1,134 @@
...
+
+//WebView manage map(game.id, panel instance)
 const openWebviews = new Map<string, vscode.WebviewPanel>();
 
+// Function for creating directories
+function ensureDirectoryExistence(directoryPath: string){
+  if (!fs.existsSync(directoryPath)) {
+    fs.mkdirSync(directoryPath, { recursive: true});
+  }
+}
```
Comment (junman95) [inline src/extension.ts]: same as comment below(line20)

File: src/extension.ts [hunk 4]
```diff
@@ -1,52 +1,134 @@
...
+
+
     }
   };
 
   let openGameCommand = vscode.commands.registerCommand(
-    'vsc-gameboy.openGame',
-    gameId => {
-      let game = metaData.find(game => game.id === gameId);
+    "vsc-gameboy.openGame",
```
Comment (junman95) [inline src/extension.ts]: We made a prettierrc.
I think your formatter seems not synchronized with our new prettier format.
pull main branch first and check it out!

File: src/extension.ts [hunk 5]
```diff
@@ -1,52 +1,134 @@
...
   // 커맨드를 확장 프로그램의 context에 추가하여 활성화 상태 유지
   context.subscriptions.push(openGameCommand);
 }
 
 // 확장 프로그램이 비활성화될 때 호출되는 메서드
-export function deactivate() {
-  // WebViewPanel 인스턴스를 모두 제거
-  openWebviews.forEach(panel => panel.dispose());
-}
+export function deactivate() {}
```
Comment (junman95) [inline src/extension.ts]: wonder why disposing webview codes are deleted

File: src/extension.ts [hunk 6]
```diff
@@ -1,52 +1,134 @@
...
+            vscode.Uri.file(
+              vscode.Uri.joinPath(
+                context.extensionUri,
+                "resource/game",
+                game.id
+              ).fsPath
+            ),
+          ],
+        }
+      );
```
Comment (junman95) [inline src/extension.ts]: Code has been reverted. Check this please

File: src/extension.ts [hunk 7]
```diff
@@ -1,52 +1,134 @@
...
       panel.webview.html = getHtmlContent(context, panel.webview, game);
 
+      handlePanelMessages(panel, context, game.id);
       openWebviews.set(game.id, panel);
 
       panel.onDidDispose(() => openWebviews.delete(game.id));
+
+
```
Comment (junman95) [inline src/extension.ts]: useless blanks.

File: src/util/get-html-content.ts [hunk 8]
```diff
@@ -13,12 +13,35 @@ function getHtmlContent(
...
+        });
+      }
+    </script>
+  `;
+    
+  htmlContent = htmlContent.replace(
+    /<\/body>/,
+    scoreScript + '</body>'
+  );
+  console.log('htmlContent',htmlContent);
```
Comment (junman95) [inline src/util/get-html-content.ts]: delete it please

File: package.json [hunk 9]
```diff
@@ -29,6 +29,10 @@
         {
           "id": "gameExplorer",
           "name": "Game Explorer"
+        },
+        {
+          "id": "scoreList",
+          "name": "score list"
```
Comment (junman95) [inline package.json]: unify convention of views!
I think capital letter on first word seems better.

**Review by junman95 — APPROVED** [review]
nice job

**Review by smshack — APPROVED** [review]
lgtm

**Pull request merged**